packages/studio-web/src/app/upload/upload.component.ts

```diff
@@ -294,7 +303,8 @@ export class UploadComponent implements OnInit {
             // We can't give the arguments types because RxJS is broken somehow,
             // see https://stackoverflow.com/questions/66615681/rxjs-switchmap-mergemap-resulting-in-obserableunknown
             this.ssjsService.align$(audio, ras as ReadAlong)
-          )
+          ),
+          takeUntil(this.unsubscribe$)
```

Comment:
Likewise - we are definitely calling the `complete` callback on the subscription here, and in fact there is an explicit unsubscribe callback that we return when creating the `Observable` which I can also confirm is getting called and destroying the decoder.